Hide zero & infinity values

Chocobo1 opened the pull request:
~~Don't merge yet!!~~

Before:
![before](https://cloud.githubusercontent.com/assets/9395168/12535392/b7ad0b10-c2bc-11e5-843d-c8a8d59b05f5.png)

After:
![after](https://cloud.githubusercontent.com/assets/9395168/12535393/baa6586c-c2bc-11e5-9936-c3b09a09b9a0.png)

Feedback are welcome!

glassez commented on src/gui/transferlistdelegate.cpp:
```diff
@@ -49,218 +51,242 @@
+}
 
-TransferListDelegate::~TransferListDelegate() {}
+TransferListDelegate::~TransferListDelegate()
```
Drop this empty destructor.

Chocobo1 replied:
Done.

glassez commented on src/gui/transferlistdelegate.cpp:
```diff
@@ -49,218 +51,242 @@
-      case BitTorrent::TorrentState::Uploading:
-      case BitTorrent::TorrentState::StalledUploading:
-        display = tr("Seeding", "Torrent is complete and in upload-only mode");
+void TransferListDelegate::paint(QPainter * painter, const QStyleOptionViewItem & option, const QModelIndex & index) const
```
This function is too long. It would be better if you split it.

Chocobo1 replied:
Done. 
The status string is not located at its own function:
`QString TransferListDelegate::getStatusString(const int state) const`

glassez replied:
I mean, even more structurally. Example:
```c++
case TorrentModel::TR_UPSPEED:
case TorrentModel::TR_DLSPEED:
      paintSpeed(/*some needed args */);
      break;
  case TorrentModel::TR_UPLIMIT:
  case TorrentModel::TR_DLLIMIT:
    paintLimit(/*some needed args */);
    break;
  }
```

birdie-github commented:
My initial implementation (pull #3557) had three options which I'd like to be implemented:

```C++
+    combo_showzeros.addItem(tr("Always show all values"));
+    combo_showzeros.addItem(tr("Only hide zeros for paused torrents"));
+    combo_showzeros.addItem(tr("Always hide zero values"));
```

Yes and no (true || false) just don't cut it for many people who've commented on the initial feature request.

ngosang commented:
Why are you hiding infinity in ETA? If  you don't show anything seems that the application can not estimate the ETA.

birdie-github commented:
Yep, I second that: Infinity should not be hidden.

Chocobo1 commented:
>My initial implementation (pull #3557) had three options which I'd like to be implemented:

Will do, but don't expect soon, I don't have much time for coding recently.

>Why are you hiding infinity in ETA? If you don't show anything seems that the application can not estimate the ETA.

>Yep, I second that: Infinity should not be hidden.

I'm not so sure what value should be hidden, so if you guys spot anything wrong, just point it out.

ngosang commented:
I think ETA column value should be displayed always, even if it's infinity.

sledgehammer999 commented:
Yes, ETA infinity. Also if you can rebase. I'll test and merge as-is for the moment. Of course, if anyone has objections about other values being hidden, speak.

Chocobo1 commented:
>I'll test and merge as-is for the moment

What? I haven't impl what birdie-github requested: https://github.com/qbittorrent/qBittorrent/pull/4675#issuecomment-174549453

I will update in a week or two...

sledgehammer999 commented:
>What? I haven't impl what birdie-github requested: #4675 (comment)

Yes, I know. Not important as a first merge. And probably people won't complain much.

DoumanAsh commented:
Ummm... why exactly it is needed? Is anyone bothered by having zeros/infinity?

sledgehammer999 commented:
@DoumanAsh read #3543.

DoumanAsh commented:
I see... i suppose world is full of strange people. Anyway since it is optional i have no complaint about implementing it, i just found it strange. Sorry for bumping in like that

Chocobo1 commented:
>My initial implementation (pull #3557) had three options which I'd like to be implemented:

Fixed.

>I think ETA column value should be displayed always, even if it's infinity.

Fixed.

Ready for testing & merging.

sledgehammer999 commented:
One minor comment. The rest looks good. I am doing a test build now to see it visually. It will be merged, just fix the minor thing.

sledgehammer999 commented:
It feels weird but, whatever. There an option to disable it.
However is it normal for the Seeds/Peers columns to not hide the `0 (0)` value?

Chocobo1 commented:
>just fix the minor thing.

It's fixed now.

sledgehammer999 commented:
Question: It always prints the `0 (0)` value for the Seeds/Peers columns. Is that normal?

Chocobo1 commented:
>Question: It always prints the 0 (0) value for the Seeds/Peers columns. Is that normal?

Now that I think of it, it should hide when both values are 0.
Fixed now.

sledgehammer999 commented:
It works. Thank you.

thalieht commented:
There are already 3 people against hiding infinity in ETA so this is probably pointless but i'm gonna say it. It's the first thing that caught my eye when i enabled this nice feature and i immediately thought it's a bug. AFAIK the only state ETA won't show infinity is when downloading and when Set Ratio Limiting is used.

a-raccoon commented:
Agreed @thalieht , either get rid of the `Hide zero and infinity` display option, or apply it to the ETA column.  I wonder if the proponents for keeping ETA displaying the infinity symbol are even using the hide option, since they seem confused by it.
